pkg/pillar/agentlog/agentlog.go

```diff
@@ -153,16 +153,17 @@ func (hook *SkipCallerHook) Levels() []logrus.Level {
 
 // Wait on channel then handle the signals
 func handleSignals(log *base.LogObject, agentName string, agentPid int, sigs chan os.Signal) {
-	for {
-		select {
-		case sig := <-sigs:
-			log.Functionf("handleSignals: received %v\n", sig)
-			switch sig {
-			case syscall.SIGUSR1:
-				dumpStacks(log, agentName)
-			case syscall.SIGUSR2:
-				go listenDebug(log, agentName)
-			}
+	agentDebugDir := fmt.Sprintf("%s/%s/", types.PersistDebugDir, agentName)
+	stacksDumpFileName := agentDebugDir + "/sigusr1"
+	memDumpFileName := agentDebugDir + "/sigusr2"
+
+	for sig := range sigs {
+		log.Functionf("handleSignals: received %v\n", sig)
+		switch sig {
+		case syscall.SIGUSR1:
+			dumpStacks(log, stacksDumpFileName)
```

Comment:
Good catch.

Reply:
@europaul may I ask you to specify the commit eb0738361433 ("agentlog: add pprof debug http handler")  in the commit message which you are fixing here.